src/main/java/com/eternalcode/discordapp/experience/listener/ExperienceMessageListener.java

```diff
@@ -12,15 +12,21 @@ public class ExperienceMessageListener extends ListenerAdapter {
 
     private final ExperienceRepository experienceRepository;
     private final ExperienceConfig experienceConfig;
+    private final ExperienceService experienceService;
 
     public ExperienceMessageListener(ExperienceRepository experienceRepository, ExperienceConfig experienceConfig) {
         this.experienceRepository = experienceRepository;
         this.experienceConfig = experienceConfig;
+        this.experienceService = new ExperienceService(this.experienceRepository);
```

Comment:
Taka inicjacja klasy nie powinna znajdować się w konstruktorze, wydziel to w głównej klasie appki